File: library/node/types/src/lib/students.ts
import { IPageInfo } from './table';

export interface IStudent {
  ID: number;
  CreatedAt: string;
  UpdatedAt: string;
  FirstName: string;
  LastName: string;
  Gender: string;
}

export interface IGetStudentsResponse {
  data: IStudent[];
  pageInfo: IPageInfo;
}

// same input with update
export interface ICreateStudentInput {
  firstName: string;
  lastName: string;
}

export interface IStudentDetails {
  ID: number;
  CreatedAt: string;
  UpdatedAt: string;
  DeletedAt: string;
  StudentId: number;
  HeightStanding: number;
  EyeHeight: number;
  ShoulderHeight: number;
  ElbowHeight: number;
  KneeHeight: number;
  SittingHeight: number;
  VerticalReachHeight: number;
  Weight: number;
  Bmi: number;
}
export interface IGetDetailsResponse {
  student: IStudent;
  data: IStudentDetails[];
}

File: library/go/models/students.go
package models

import "gorm.io/gorm"

type Student struct {
	gorm.Model
	FirstName     string `gorm:"not null"`
	LastName      string `gorm:"not null"`
	Gender        string `gorm:"not null"`
	StudentDetail []StudentDetail
}

type StudentDetail struct {
	gorm.Model
	StudentId           uint `gorm:"not null"`
	HeightStanding      float64
	EyeHeight           float64
	ShoulderHeight      float64
	ElbowHeight         float64
	KneeHeight          float64
	SittingHeight       float64
	VerticalReachHeight float64
	Weight              float64
	Bmi                 float64
}
